### DISCUSSION TITLE
Behavior of absolute value, under, and complex numbers

### DISCUSSION THREAD

**Discussion by Omnikar** [uiua-lang/uiua#261] [Language Development]
There is an intuitive behavior of using `⌵ absolute value` and `⍜ under` together on a real number; that is, applying the function given to `under` to the positive version of the number, but then leaving it with its original sign. I think the sensible analogue of this for complex numbers would be to maintain the direction of the original complex number, while scaling it to fit the new magnitude outputted by the function given to `under`. Whatever the case, however, the current behavior of combining `under`, `absolute value`, and complex numbers is certainly nonsensical.

For example, pass `∘ identity` as the second function to `⍜ under`:
```
⍜⌵∘ ℂ`3 4
```
This code will output `5+¯5i` to the stack. As demonstrated, `⍜⌵∘` maintains the sign of each component of the complex number, but replaces both components with the magnitude of the original number.

**Comment by kaikalii** [comment]
Fixed in `main` and the site.
`⌵ absolute value` now normalizes complex numbers.

**Comment by kaikalii** [comment]
It will work with `under` too.
